// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_package = "moniter.grpc";

package moniter;

// Computer status monitering service
service MoniterService {
  rpc current_memory_moniter_method(MemoryRequest) returns (MemoryReply) {}
}


// The request message containing client order
message MemoryRequest {
  string virtual_memory_request = 1;
  string physical_memory_request = 2;
  string avail_virtual_memory_request = 3;
  string avail_physical_memory_request = 4;
}

// The response message containing the information of memory system
message MemoryReply {
  string memory_info_reply = 1;
}
